MovingBlocks/DestinationSol
# NUI talk screen

# Description
This pull request converts the Talk screen to NUI. It should functionally identically to before, although there may be some minor visual differences.

### Before
![image](https://user-images.githubusercontent.com/24301287/142293458-125d1809-038c-4842-b9cc-f082a491af3c.png)

### After
![image](https://user-images.githubusercontent.com/24301287/142293579-9837ce0b-0d96-408f-b061-5ec86f0cc056.png)

# Testing
- Start a new game (or continue an existing one)
- Fly towards the starting station until the `Talk` button is enabled
- Verify that the talk screen opens when you click on the `Talk` button
- Verify that each of the buttons on the talk screen behaves as before e.g.
  - Verify that the buy screen opens when you click on the `Buy` button (or press the `B` key)
  - Verify that the sell screen opens when you click on the `Sell` button (or press the `S` key)
  - Verify that the `Change Ship` screen opens when you click on the `Change Ship` button (or press the `C` key)
  - Verify that the `Hire Ship` screen opens when you click on the `Hire` button (or press the `H` key)
  - Verify that the talk screen closes when you click on its `Close` button (or press the `Escape` key)
- Test that this works for both the starting station and the stations found on planets

# Notes
- This depends on #631, #636 and #638.
- Until #631, #636 and #638 are merged, it might be easier to view the changes made in this pull request by viewing the diff between this and #638 at https://github.com/BenjaminAmos/DestinationSol/compare/nui-game-menu...nui-talk-screen.
